crates/common/fork_choice/src/store.rs

```diff
@@ -295,4 +306,286 @@ impl Store {
+        &mut self,
+        attestation: &Attestation,
+        is_from_block: bool,
+    ) -> anyhow::Result<()> {
+        let target = attestation.data.target;
+
+        // If the given attestation is not from a beacon block message, we have to check the target
+        // epoch scope.
+        if !is_from_block {
+            self.validate_target_epoch_against_current_time(attestation)?;
+        }
+
+        // Check that the epoch number and slot number are matching
+        ensure!(target.epoch == compute_epoch_at_slot(attestation.data.slot));
+
+        // Attestation target must be for a known block. If target block is unknown, delay
+        // consideration until block is found
+        ensure!(self.blocks.contains_key(&target.root));
+
+        // Attestations must be for a known block. If block is unknown, delay consideration until
+        // the block is found
+        ensure!(
+            self.blocks
+                .contains_key(&attestation.data.beacon_block_root)
+        );
+        // Attestations must not be for blocks in the future. If not, the attestation should not be
+        // considered
+        ensure!(self.blocks[&attestation.data.beacon_block_root].slot <= attestation.data.slot);
+
+        // LMD vote must be consistent with FFG vote target
+        ensure!(
+            target.root
+                == self.get_checkpoint_block(attestation.data.beacon_block_root, target.epoch)?
+        );
+
+        // Attestations can only affect the fork choice of subsequent slots.
+        // Delay consideration in the fork choice until their slot is in the past.
+        ensure!(self.get_current_slot() >= attestation.data.slot + 1);
+
+        Ok(())
+    }
+
+    pub fn store_target_checkpoint_state(&mut self, target: Checkpoint) -> anyhow::Result<()> {
+        // Store target checkpoint state if not yet seen
+        if let Entry::Vacant(entry) = self.checkpoint_states.entry(target) {
+            let mut base_state = self.block_states[&target.root].clone();
+            let target_slot = compute_start_slot_at_epoch(target.epoch);
+            if base_state.slot < target_slot {
+                base_state.process_slots(target_slot)?;
+            }
+            entry.insert(base_state);
+        }
+
+        Ok(())
+    }
+
+    pub async fn is_data_available(
+        &self,
+        blob_kzg_commitments: &[KZGCommitment],
+        execution_engine: &impl ExecutionApi,
+        beacon_block_root: B256,
+    ) -> anyhow::Result<bool> {
+        // `retrieve_blobs_and_proofs` is implementation and context dependent
+        // It returns all the blobs for the given block root, and raises an exception if not
+        // available Note: the p2p network does not guarantee sidecar retrieval outside of
+        // `MIN_EPOCHS_FOR_BLOB_SIDECARS_REQUESTS`
+        let blob_count = blob_kzg_commitments.len();
+        let mut blobs_and_proofs: Vec<Option<BlobsAndProofV1>> = vec![None; blob_count];
```

Comment:
```suggestion
        let mut blobs_and_proofs: Vec<Option<BlobsAndProofV1>> = vec![None; blob_kzg_commitments.len()];
```
We don't need to define a var anymore as it is only used once